### PR TITLE
Documentation v1

### PR DESCRIPTION
**Pull request by dprokop**
TODO: 
- [x] Handling query statuses
- [x] Caching
- [x] Server-side rendering
- [x] API
- [x] Known limitations

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,3 +3,177 @@
...
+Querier is a tool that helps you get remote and async data into React app effortlessly.
+
+With simple, declarative API, you don't have to worry about handling async state or errors. All you have to do is decorate component that requires async data with `withData` higher-order component. Then, results, states and possible errors will be injected into the component as props.
+
+Querier is implemented and tested in TypeScript.
+
+## What is a declarative data layer?
+Simply, it is a way to supply data to your components. You declare component's data requirements, and the whole logic for fetching it, supplying status and results is handled by Querier itself. To achieve that, you implement `queries`. Queries are simply **async functions**. They can either communicate with remote source or resolve async data in whatever way you imagine.
+
+## Querier Gist
```
Comment (konrad147) [inline README.md:30]: Sample is written in Typescript. It may be worth to add some comment about JS usage.

File: README.md [hunk 2]
```diff
@@ -3,3 +3,177 @@
...
+import {  withData,  WithDataProps,  QuerierState } from 'querier';
+
+type RepositoryDetailsOwnProps = {  id: number }
+type RepositorDetailsInputQueries = {  repository: RepositoryType }
+type RepositorDetailsActionQueries = {  starRepository: RepositoryType }
+type RepositoryDetailsProps = WithDataProps<
+  RepositoryDetailsOwnProps,
+  RepositoryDetailsInputQueries,
+  RepositorDetailsActionQueries
+>;
```
Comment (konrad147) [inline README.md]: Maybe some comment why are those types needed for

File: README.md [hunk 3]
```diff
@@ -3,3 +3,177 @@
...
+
+class SearchComponent extends React.Component<WithDataProps<{}, {} , SearchComponentActionQueries>> {
+  handleSubmit(userName: string) {
+    if (this.props.actionQueries && this.props.actionQueries.searchUser) {
+      this.props.actionQueries.searchUser(userName);
+    }
+  }
+
+  render () {
+    // ...
```
Comment (konrad147) [inline README.md]: Sample of using actionQuery state?

File: README.md [hunk 4]
```diff
@@ -3,3 +3,177 @@
...
+      query: getRepository
+    }
+  }
+}
+
+export default withData(repositoryDetailsQueries)(YourComponent);
+```
+Input queries are fired when component mounts. **They always receive component's full set of props** as argument. Later, they are being reinvoked when component's props change.
+
+Input queriy results are **cached** by default. Please see [Caching]() for more details.
```
Comment (konrad147) [inline README.md]: typo :)

File: README.md (hunk 3 above)
Comment (dprokop) [reply README.md]: This is still on TODO list. Gonna write about it soon.

File: README.md [hunk 5]
```diff
@@ -3,3 +3,177 @@
 Simple declarative data layer for React applications
 
 [![Build Status](https://travis-ci.org/dprokop/querier.svg?branch=master)](https://travis-ci.org/dprokop/querier)
+
+# Table of Content
+  * [What is Querier?](#what-is-querier)
+  * [What is a declarative data layer?](#what-is-declarative-data-layer)
```
Comment (shpackus) [inline README.md]: This link is broken. Missing `-a-` in anchor definition I guess

File: README.md (hunk 1 above)
Comment (aniaw) [reply README.md:30]: Is it possible to shorten/simplify this gist? At first glance, it is hard for me to find `querier` things.

Comment (dprokop) [reply README.md:30]: Good point. I think I'll simply write it in pure JS for matter of simplicity

File: README.md (hunk 3 above)
Comment (dprokop) [reply README.md]: @konrad147 ready in 7f2c1eee5b7081eb6cac1dfc6071ed439efa8c69

File: README.md (hunk 2 above)
Comment (dprokop) [reply README.md]: Ready in 7f2c1eee5b7081eb6cac1dfc6071ed439efa8c69. Btw according to @aniaw comment I'm gonna probably simplify this example and use pure JS.

File: README.md (hunk 1 above)
Comment (dprokop) [reply README.md:30]: Fixed in a0ee31bef80866d0cb91a525f23c4ffbd8764359

**Pull request merged**